eureka-client/src/main/java/com/netflix/discovery/shared/resolver/ClusterResolver.java

```diff
@@ -21,9 +21,9 @@
 /**
  * @author Tomasz Bak
  */
-public interface ClusterResolver {
+public interface ClusterResolver<T extends EurekaEndpoint> {
```

Comment:
Do we want to have extended EurekaEndpoint interfaces, or just different implementations? In the latter case, I would drop the generic type parameter and use straight EurekaEndpoint as previously.